JCenter is decommissioning May 1, 2021. Replace with MavenCentral repo.

### Summary

JCenter is being decommissioned in 6 weeks. Gradle repositories should be replaced with Maven Central sources.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android, iOS, Web

### SDK Version (managed workflow only)

_No response_

### Environment

 Expo CLI 4.1.6 environment info:
    System:
      OS: Windows 10 10.0.19042
    Binaries:
      Node: 14.15.5 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.10 - C:\Program Files\nodejs\yarn.CMD
      npm: 6.14.11 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ^40.0.1 => 40.0.1
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: 0.63.4 => 0.63.4
    Expo Workflow: bare

### Reproducible demo or steps to reproduce from a blank project

This will not be reproducible until May 1, 2021 when JCenter requests begin failing. We should transition before this happens.

JCenter [is being decommissioned in 6 weeks](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/). Gradle repositories should be replaced with Maven Central sources. Flipper has already begun its transition.

Thanks, @brentvatne 

A [build of mine](https://expo.dev/accounts/wodin/projects/expo-ocr-example/builds/90ab0832-2df0-410c-987b-97d47047f3df) just failed with the following error:

`Could not GET 'https://jcenter.bintray.com/com/facebook/react/react-native/maven-metadata.xml'. Received status code 502 from server: Bad Gateway`

```
[stderr] FAILURE: Build failed with an exception.
[stderr] * What went wrong:
[stderr] Could not determine the dependencies of task ':app:lintVitalRelease'.
[stderr] > Could not resolve all artifacts for configuration ':app:debugRuntimeClasspath'.
[stderr]    > Could not resolve com.facebook.react:react-native:+.
[stderr]      Required by:
[stderr]          project :app
[stderr]          project :app > project :expo
[stderr]          project :app > project :react-native-svg
[stderr]          project :app > project :react-native-tesseract-ocr
[stderr]          project :app > project :expo > project :expo-modules-core
[stderr]          project :app > project :expo > project :expo-dev-launcher
[stderr]          project :app > project :expo > project :expo-dev-menu
[stderr]          project :app > project :expo > project :expo-dev-menu-interface
[stderr]          project :app > project :expo > project :expo-splash-screen
[stderr]       > Failed to list versions for com.facebook.react:react-native.
[stderr]          > Unable to load Maven meta-data from https://jcenter.bintray.com/com/facebook/react/react-native/maven-metadata.xml.
[stderr]             > Could not get resource 'https://jcenter.bintray.com/com/facebook/react/react-native/maven-metadata.xml'.
[stderr]                > Could not GET 'https://jcenter.bintray.com/com/facebook/react/react-native/maven-metadata.xml'. Received status code 502 from server: Bad Gateway
```

If I try loading that URL in a browser I get the following:
```
Error: Server Error
The server encountered a temporary error and could not complete your request.

Please try again in 30 seconds.
```

Fortunately I was able to workaround it by removing all references to `jcenter()` in all of the `build.gradle` files using a Config Plugin and `patch-package`:

https://github.com/wodin/expo-ocr-example/commit/f8967575bd6d5c950edd148b3424b94728b72100

--- a/packages/@expo/cli/src/prebuild/__tests__/fixtures/react-native-project.ts
+++ b/packages/@expo/cli/src/prebuild/__tests__/fixtures/react-native-project.ts
@@ -1069,7 +1069,6 @@ export default {
           }
           repositories {
               google()
-              jcenter()
           }
           dependencies {
               classpath("com.android.tools.build:gradle:3.5.3")
@@ -1092,7 +1091,6 @@ export default {
               }
       
               google()
-              jcenter()
               maven { url 'https://www.jitpack.io' }
           }
       }`,

--- a/packages/@expo/config-plugins/src/android/__tests__/GoogleServices-test.ts
+++ b/packages/@expo/config-plugins/src/android/__tests__/GoogleServices-test.ts
@@ -95,7 +95,6 @@ buildscript {
     }
     repositories {
         google()
-        jcenter()
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.5.3'
@@ -111,7 +110,6 @@ buildscript {
     }
     repositories {
         google()
-        jcenter()
     }
     dependencies {
         classpath 'com.google.gms:google-services:4.4.1'

